lib/trento_web/controllers/v1/discovery_controller.ex

```diff
@@ -1,21 +1,48 @@
 defmodule TrentoWeb.V1.DiscoveryController do
   use TrentoWeb, :controller
+  use OpenApiSpex.ControllerSpecs
 
   alias Trento.Integration.Discovery
 
+  alias TrentoWeb.OpenApi.Schema
+
+  plug OpenApiSpex.Plug.CastAndValidate, json_render_error_v2: true
   action_fallback TrentoWeb.FallbackController
 
-  @spec collect(Plug.Conn.t(), map) :: Plug.Conn.t()
-  def collect(conn, event) do
+  operation :collect,
+    summary: "Collect data from the target infrastructure",
+    tags: ["Agent"],
+    description:
+      "This endpoint is used by the agents to collect data from the target infrastructure",
+    request_body: {"Checks Selection", "application/json", Schema.DiscoveryEvent},
```

Comment:
`Checks Selection`? This should be `Discovery Event` or anything like that, right?

Reply:
yep!